test/cpp/end2end/xds/xds_fallback_end2end_test.cc

```diff
@@ -90,18 +91,32 @@ class XdsFallbackTest : public XdsEnd2endTest {
         cluster.eds_cluster_config().service_name()));
   }
 
-  void ConfigureAuthority(const std::string& eds_server_name,
-                          const std::string& cluster_name,
-                          const std::string& route_config_name,
-                          const std::string& listener_name,
-                          const EdsResourceArgs& args) {
-    fallback_balancer_->ads_service()->SetEdsResource(
-        BuildEdsResource(args, eds_server_name));
+  void ConfigureAuthority(BalancerServerThread* balancer,
```

Comment:
I think we should combine this with the existing `SetXdsResourcesForTarget()` method, as you had originally started to do:

```
  void SetXdsResourcesForServer(BalancerServerThread* balancer, size_t backend,
                                absl::string_view server_name = "",
                                absl::string_view authority = "") {
    Listener listener = default_listener_;
    RouteConfiguration route_config = default_route_config_;
    Cluster cluster = default_cluster_;
    // Default server uses default resources when no authority, to enable using more
    // test framework functions.
    if (!server_name.empty() || !authority.empty()) {
      auto get_resource_name =
          [&](absl::string_view resource_type) {
            if (authority.empty()) {
              if (resource_type == kLdsTypeUrl) return std::string(server_name);
              return absl::StrFormat("%s_%s", resource_type, server_name);
            }
            return absl::StrFormat("xdstp://%s/%s/%s", authority, resource_type, server_name);
          };
      listener.set_name(get_resource_name(kLdsTypeUrl));
      cluster.set_name(get_resource_name(kCdsTypeUrl);
      cluster.mutable_eds_cluster_config()->set_service_name(get_resource_name(kEdsTypeUrl);
      route_config.set_name(get_resource_name(kRdsTypeUrl));
      route_config.mutable_virtual_hosts(0)
          ->mutable_routes(0)
          ->mutable_route()
          ->set_cluster(cluster.name());
    }
    SetListenerAndRouteConfiguration(balancer, listener, route_config);
    balancer->ads_service()->SetCdsResource(cluster);
    balancer->ads_service()->SetEdsResource(BuildEdsResource(
        EdsResourceArgs(
            {{"locality0", CreateEndpointsForBackends(backend, backend + 1)}}),
        cluster.eds_cluster_config().service_name()));
  }
```

Note that this approach will avoid the config tear that you're handling in the `PerAuthorityFallback` test below, because the resource names will be the same on the primary and secondary servers.

Also note that I went back to using the term "server" instead of "target" here, since in the authority case, the string is not the full target URI.

Reply:
Done.